doxygen.cmake

```diff
@@ -184,6 +184,28 @@ MACRO(_SETUP_PROJECT_DOCUMENTATION_FINALIZE)
       SET(DOXYGEN_HEADER_NAME "header.html")
     ENDIF()
 
+    # Find doxytag files
+    # To ignore this list of tag files, add to doc/Doxyfile.extra.in
+    # TAGFILES =
+    STRING(REPLACE "," ";" PKG_REQUIRES "${_PKG_CONFIG_REQUIRES}")
+    FOREACH(PKG_CONFIG_STRING ${PKG_REQUIRES})
+      # Get the name of the package in the same format as in pkg-config.cmake.
+      STRING(REGEX MATCH "[^<>= ]+" LIBRARY_NAME "${PKG_CONFIG_STRING}")
+      STRING(REGEX REPLACE "[^a-zA-Z0-9]" "_" PREFIX "${LIBRARY_NAME}")
+      STRING(TOUPPER "${PREFIX}" "PREFIX")
+      # If DOXYGENDOCDIR is specified, add a doc path.
+      IF( DEFINED ${PREFIX}_DOXYGENDOCDIR
+          AND EXISTS ${${PREFIX}_DOXYGENDOCDIR}/${LIBRARY_NAME}.doxytag)
+        IF(DEFINED _PKG_CONFIG_DOXYGENDOCDIR)
+          FILE(RELATIVE_PATH DOXYTAG_FILE ${_PKG_CONFIG_DOXYGENDOCDIR} ${${PREFIX}_DOXYGENDOCDIR})
+        ELSE(DEFINED _PKG_CONFIG_DOXYGENDOCDIR)
```

Comment:
I'd rather we don't add repeated conditions in new code in the modules. They are the ghosts of cmake past 👻

Reply:
I guess variable `_PKG_CONFIG_DOXYGENDOCDIR` is always defined when we reach this point.